browser/src/CopyNumberVariantList/CopyNumberVariants.tsx

```diff
@@ -0,0 +1,309 @@
+  exportFileName: string
+  variants: CopyNumberVariantPropType[]
+}
+
+const CopyNumberVariants = ({ context, exportFileName, variants }: CopyNumberVariantsProps) => {
+  const table = useRef(null)
+  const tracks = useRef(null)
+
+  const [selectedColumns, setSelectedColumns] = useState(() => {
+    try {
+      return userPreferences.getPreference('copyNumberVariantTableColumns') || DEFAULT_COLUMNS
+    } catch (error) {
+      return DEFAULT_COLUMNS
+    }
+  })
+
+  const renderedTableColumns = useMemo(() => {
+    const columnsForContext = getColumnsForContext(context)
+    return (
+      ['variant_id', ...selectedColumns]
+        // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
+        .map((columnKey) => columnsForContext[columnKey])
+        .filter(Boolean)
+        .map((column) => ({
+          ...column,
+          isSortable: Boolean(column.compareFunction),
+          tooltip: column.description,
+        }))
+    )
+  }, [context, selectedColumns])
+
+  const [filter, setFilter] = useState({
+    includeTypes: {
+      DEL: true,
+      DUP: true,
+    },
+    includeFilteredVariants: false,
+    searchText: '',
+  })
+
+  const [sortState, setSortState] = useState({
+    sortKey: 'variant_id',
+    sortOrder: 'ascending',
+  })
+  const { sortKey, sortOrder } = sortState
+
+  // @ts-expect-error TS(7006) FIXME: Parameter 'newSortKey' implicitly has an 'any' typ... Remove this comment to see the full error message
+  const setSortKey = useCallback((newSortKey) => {
+    setSortState((prevSortState) => {
+      if (newSortKey === prevSortState.sortKey) {
+        return {
+          sortKey: newSortKey,
+          sortOrder: prevSortState.sortOrder === 'ascending' ? 'descending' : 'ascending',
+        }
+      }
+
+      return {
+        sortKey: newSortKey,
+        sortOrder: 'descending',
+      }
+    })
+  }, [])
+
+  const filteredVariants = useMemo(
+    () => filterCopyNumberVariants(variants, filter, renderedTableColumns),
+    [variants, filter, renderedTableColumns]
+  )
+
+  const renderedVariants = useMemo(
+    () => sortVariants(filteredVariants, sortState),
+    [filteredVariants, sortState]
+  )
+
+  const [showTableConfigurationModal, setShowTableConfigurationModal] = useState(false)
+  const [variantHoveredInTable, setVariantHoveredInTable] = useState(null)
+  const [variantHoveredInTrack, setVariantHoveredInTrack] = useState(null)
+
+  const shouldHighlightTableRow = useCallback(
+    // @ts-expect-error TS(7006) FIXME: Parameter 'variant' implicitly has an 'any' type.
+    (variant) => {
+      return variant.variant_id === variantHoveredInTrack
+    },
+    [variantHoveredInTrack]
+  )
+
+  // @ts-expect-error TS(7031) FIXME: Binding element 'scrollOffset' implicitly has an '... Remove this comment to see the full error message
+  const onScrollTable = useCallback(({ scrollOffset, scrollUpdateWasRequested }) => {
```

Comment:
As a rule, don't name callback functions for the callback they're attached to. Rather, use the name to describe what they do, as you would with any other function. For instance, here you might call the function `scrollToCurrentTrack`.